spec/report_spec.rb

```diff
@@ -34,14 +34,19 @@
     module_results = report.results['module.rb']
     # only includes changed methods
     expect(module_results.map(&:name))
-      .to eq(%w[BaconModule bar baz])
+      .to eq(%w[BaconModule bar baz branch_missed])
 
     # includes flagged blocks
     module_flagged = module_results.select(&:flagged?)
-    expect(module_flagged.size).to eq(1)
+    expect(module_flagged.size).to eq(2)
     expect(module_flagged[0].node.name).to eq('bar')
     expect(module_flagged[0].coverage_f).to eq(0.0)
 
+    # includes flagged blocks with incomplete branch coverage
+    expect(module_flagged[1].node.name).to eq('branch_missed')
+    pending 'coverage_f need to handle branch coverage data now present in Result'
```

Comment:
The `pending` should be removed before we merged - right now the `coverage_f` for `branch_missed` fixture is > 0 (L48)

Reply:
OK that's another good catch. I made a WIP commit that refactors `coverage_f`. I still need to write some specs for that new algorithm but I don't fully understand the previous implementation or even the feature yet. So probably I will get to that next weekend...

Reply:
Cool, I'd be happy to provide any help if you need it. `coverage_f` returns the "classic" line coverage percentage for a given `Result` (representing a method, class, ruby block).

Reply:
Thanks, I understand it now.
I was a bit confused by the existing implementation, but I believe my refactor should be correct now.